pkg/pod/status.go

```diff
@@ -125,8 +125,8 @@ func MakeTaskRunStatus(tr v1alpha1.TaskRun, pod *corev1.Pod, taskSpec v1alpha1.T
 				msg := s.State.Terminated.Message
 				r, err := termination.ParseMessage(msg)
 				if err != nil {
-					logger.Errorf("Could not parse json message %q because of %w", msg, err)
-					break
+					// step generated non-JSON termination message; ignore
+					continue
```

Comment:
this change seems a bit inconsistent with the other error handling in this function, which will still be logging and `break`ing, 2 thoughts:

1. are we sure we want to continue processing messages when this happens? altho we dont want to panic, maybe it's reasonably to assume that encountering a badly formatted message indicates the rest are bogus, or is it that we could have some steps with good termination messages, and some with bad?
2. maybe logging here still makes sense? if you want to track down what happened later?

Reply:
side note / scope creep: why are we creating a new logger in this function? if this is in the controller, can we pass in the (properly configured) logger instead? <-- probably the material for a separate issue

Reply:
> this change seems a bit inconsistent with the other error handling in this function, which will still be logging and breaking

This is true, though the other `break`s in this function, lines 136 & 148, are inside another nested `for` loop.  That means when those lines are hit the outer `for` loop is allowed to continue on processing the rest of the steps.

> are we sure we want to continue processing messages when this happens? altho we dont want to panic, maybe it's reasonably to assume that encountering a badly formatted message indicates the rest are bogus, or is it that we could have some steps with good termination messages, and some with bad?

My feeling is that this issue is evidence of `some steps with good termination messages, and some with bad`. There are other ways that a Step might inadvertently break the termination message and I think we can afford to be a little more robust in that scenario. Off the top of my head one example where this could happen again would be that a process spawned inside the container could race with the entrypoint to write the termination log and clobber it. That would be some awful user error but it isn't inconceivable, and I think Tekton should be able to handle that gracefully without abandoning the rest of the steps.

> maybe logging here still makes sense? if you want to track down what happened later?

Yeah that's totally fair.  I'll add this back in and look at threading the Logger through from the caller.

Reply:
I've threaded the logger through from the controller and added the log line back in.

Reply:
I've factored the inner for loop into its own function.  I think this makes the control flow a little clearer.  I've also added a test for the panic-inducing case.